experiments/ClimaEarth/user_io/arg_parsing.jl

```diff
@@ -1,5 +1,21 @@
 import YAML
 
+abstract type AbstractModeType end
```

Comment:
We could change `ModeType` -> `SimMode` here and throughout (e.g. `mode_type` would become `sim_mode` too). What do you think?

Reply:
`SimulationMode`

Reply:
move these to Interfacer module